pkg/kubelet/dockershim/network/kubenet/kubenet_linux.go

```diff
@@ -464,44 +588,73 @@ func (plugin *kubenetNetworkPlugin) TearDownPod(namespace string, name string, i
 		return fmt.Errorf("Kubenet needs a PodCIDR to tear down pods")
 	}
 
-	// no cached IP is Ok during teardown
-	podIP, _ := plugin.podIPs[id]
-	if err := plugin.teardown(namespace, name, id, podIP); err != nil {
+	if err := plugin.teardown(namespace, name, id); err != nil {
 		return err
 	}
 
 	// Need to SNAT outbound traffic from cluster
 	if err := plugin.ensureMasqRule(); err != nil {
 		klog.Errorf("Failed to ensure MASQ rule: %v", err)
 	}
-
 	return nil
 }
 
 // TODO: Use the addToNetwork function to obtain the IP of the Pod. That will assume idempotent ADD call to the plugin.
 // Also fix the runtime's call to Status function to be done only in the case that the IP is lost, no need to do periodic calls
 func (plugin *kubenetNetworkPlugin) GetPodNetworkStatus(namespace string, name string, id kubecontainer.ContainerID) (*network.PodNetworkStatus, error) {
-	plugin.mu.Lock()
```

Comment:
why is the lock grabbing here removed?

Reply:
it is more granular now:
- the lock is held in `event` case
- while updating the in-mem pod ips
- when reading pod ips in `getNetworkStatus`